cmd/imagetool/getFileInImage.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/Symantec/Dominator/lib/filesystem"
+	"github.com/Symantec/Dominator/lib/fsutil"
+	objectclient "github.com/Symantec/Dominator/lib/objectserver/client"
+)
+
+func getFileInImageSubcommand(args []string) {
+	_, objectClient := getClients()
+	var outFileName string
+	if len(args) > 2 {
+		outFileName = args[2]
+	}
+	err := getFileInImage(objectClient, args[0], args[1], outFileName)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting file in image: %s\n", err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func getFileInImage(objectClient *objectclient.ObjectClient, imageName,
+	imageFile, outFileName string) error {
+	fs, err := getTypedImage(imageName)
+	if err != nil {
+		return err
+	}
+	filenameToInodeTable := fs.FilenameToInodeTable()
+	if inum, ok := filenameToInodeTable[imageFile]; !ok {
+		return fmt.Errorf("file: \"%s\" not present in image", imageFile)
+	} else if inode, ok := fs.InodeTable[inum]; !ok {
```

Comment:
OK this is what I meant:
you have
if !$precondition1
     return
else if !precondition2
     return
else if !precondition3
   return

What I am stating is that he else is unneeded and this is just
if !precondition1 
    return
if !precondition2
   return
if !precondition3
   return

Which is easier to read.

Reply:
Oh. So, you mean move the code block starting with:
`size, reader, err := objectClient.GetObject(inode.Hash)`
outside the `else` statement?

That won't compile, because `inode` is defined inside the scope of the conditionals. It can't be defined outside the scope of the conditionals because validity checks are needed first.

Reply:
define a the varible before you use it?

Reply:
No, because the type is changing as the code drills down to be more specific.

Reply:
I think we can get around this, but I am not blocking this anymore

Reply:
Thanks. I think it would be more cumbersome to do it differently, though.